docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,139 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
-        .collect(
-            Collectors.toMap(
-                e -> e.arg2,
-                e -> e.arg3
-            )
-        );
-
-    ExampleEnumInit(int arg1, String arg2, boolean arg3) {
-        // ...
-    }
-
-    // Matches the constructor
-    public static ExampleEnumInit create(String name, int arg1, String arg2, boolean arg3) {
-        throw new IllegalStateException("Enum not extended");
-    }
-
-    @Override
-    public void init() {
-        // Gets called for every new enum
-        ARG2_TO_ARG3.put(this.arg2, this.arg3);
+public class MyEnumParams {
+    public static final EnumProxy<Rarity> CUSTOM_RARITY_ENUM_PROXY = new EnumProxy<>(
+            Rarity.class, -1, "examplemod:custom", (UnaryOperator<Style>) style -> style.withItalic(true)
+    );
+    
+    public static Object getTestDamageEffectsParameter(int idx, Class<?> type) {
+        return type.cast(switch (idx) {
+            case 0 -> "examplemod:test";
+            case 1 -> (Supplier<SoundEvent>) () -> SoundEvents.DONKEY_ANGRY;
+            default -> throw new IllegalArgumentException("Unexpected parameter index: " + idx);
+        });
     }
 }
 ```
 
-### Codecs
+#### Constructor
 
-[Enum codecs][codec] generally take in the `values` array and resolve the codec instantly. As the array is evaluated before any mod entries are added, no mod entries will be supported. There are two solutions to this problem, which to choose depends on whether the enum implements the `StringRepresentable` interface or not.
+The constructor must be specified as a method descriptor and must only contain the parameters visible in the source code, omitting the hidden constant name and ordinal parameters.
 
-For normal, non-`StringRepresentable` enums, the codec can be wrapped via `Codec#lazyInitialized` or [`NeoForgeStreamCodecs#lazy`][streamcodec]. These prevent the codec from being resolved until first usage, which will always be after all mod entries are added.
+#### Parameters
 
-```java
-// For some enum with codec
-public enum ExampleEnumCodec {
-    // ...
-    ;
-
-    public static final Codec<ExampleEnumCodec> CODEC = Codec.of(/* ... */);
-    public static final StreamCodec<ByteBuf, ExampleEnumCodec> CODEC = StreamCodec.of(/* ... */);
+The parameters can be specified in three ways with limitations depending on the parameter types:
 
-    // ...
-}
+- Inline in the JSON file as an array of constants (only allowed for primitive values, Strings and for passing null to any reference type)
+- As a reference to a field of type `EnumProxy<TheEnum>` in a class from the mod (see `EnumProxy` example above)
+- As a reference to a method returning `Object`, where the return value is the parameter value to use. The method must have exactly two parameters of type `int` (index of the parameter) and `Class<?>` (expected type of the parameter)
+  - The `Class<?>` object should be used to cast (`Class#cast()`) the return value in order to keep `ClassCastException`s in mod code.
```

Comment:
I would order these how you order them in the above example. Or, it would probably be better to construct a separate example representing how each is used, because this feels a bit too brief.